Sources/Data/models/FilterListSetting.swift

```diff
@@ -0,0 +1,105 @@
+// Copyright 2022 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import CoreData
+import Shared
+
+private let log = Logger.browserLogger
+
+public final class FilterListSetting: NSManagedObject, CRUD {
+  /// The directory to which we should store all the dowloaded files into
+  private static var filterListBaseFolderURL: URL? {
+    let location = FileManager.SearchPathDirectory.applicationSupportDirectory
+    return FileManager.default.urls(for: location, in: .userDomainMask).first
+  }
+  
+  @NSManaged public var uuid: String
+  @NSManaged public var componentId: String?
+  @NSManaged public var isEnabled: Bool
+  @NSManaged private var folderPath: String?
+
+  public var folderURL: URL? {
+    get {
+      return Self.makeFolderURL(forFilterListFolderPath: folderPath)
+    }
+    set {
+      // We need to extract the path. We don't want to store the full URL
+      self.folderPath = Self.extractFolderPath(fromFilterListFolderURL: newValue)
+    }
+  }
+  
+  /// Load all the flter list settings
+  public class func loadAllSettings(fromMemory: Bool) -> [FilterListSetting] {
+    return all(context: fromMemory ? DataController.viewContextInMemory : DataController.viewContext) ?? []
+  }
+  
+  /// Create a filter list setting for the given UUID and enabled status
+  public class func create(uuid: String, componentId: String?, isEnabled: Bool, inMemory: Bool) -> FilterListSetting {
+    var newSetting: FilterListSetting!
+
+    // Settings are usually accesed on view context, but when the setting doesn't exist,
+    // we have to switch to a background context to avoid writing on view context(bad practice).
+    let writeContext = inMemory ? DataController.newBackgroundContextInMemory() : DataController.newBackgroundContext()
+
+    save(on: writeContext) {
```

Comment:
This function blocks the current thread (main thread) waiting on a `background` task to complete.
It calls `performAndWait`.

Reply:
I only copied what we do in other places. I couldn't figure out how to use the background context as it crashes every time I use it.